File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> q = new PriorityQueue<>();
        for (int s : scoville) {
            q.add(s);
        }

        int answer = 0;

        while (q.size() >= 2) {
            Integer first = q.poll();
            if (first >= K) {
                return answer;
            }

            int hotter = first + q.poll() * 2;
            q.add(hotter);
            answer++;
        }

        return (q.poll() >= K) ? answer : -1;
    }
}

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)
    answer = 0
    
    while len(scoville) >= 2:
        first = heapq.heappop(scoville)
        
        if first >= K:
            return answer
            
        second = heapq.heappop(scoville)
        heapq.heappush(scoville, (first + second * 2))
        answer += 1
    
    return answer if heapq.heappop(scoville) >= K else -1